weldx/time.py

```diff
@@ -492,29 +498,67 @@ def _convert_other(time) -> Union[pd.TimedeltaIndex, pd.DatetimeIndex]:
             f"to pd.DatetimeIndex or pd.TimedeltaIndex"
         )
 
-    @staticmethod
-    def union(times=List[Union[types_time_like, "Time"]]) -> Time:
-        """Calculate the union of multiple `Time` instances (or supported objects).
+    class _UnionDescriptor:
+        """Enables different behavior of `.union` as class and instance method."""
 
-        Any reference time information will be dropped.
+        def __get__(self, ins, typ):
+            if ins is None:
+                return typ._union_class
+            return ins._union_instance
 
-        Parameters
-        ----------
-        times
-            A list of time class instances
+    # todo: Sphinx renders the docstring below as expected. However, the examples are
+    #       not run by doctest. A possible solution in Python 3.9 can be found here:
+    #       https://stackoverflow.com/a/8820636/6700329
+    union = _UnionDescriptor()
+    """Calculate the union of multiple time-like objects.
 
-        Returns
-        -------
-        Time
-            The time union
+    This method can eiter be used as a class or instance method. When used on an
+    instance, its values are included in the calculated time union.
 
-        """
+    Note that any reference time information will be dropped.
+
+    Parameters
+    ----------
+    times: Sequence[types_time_like]
+        A list of time-like objects
+
+    Returns
+    -------
+    Time
+        The time union
+
+    Examples
+    --------
+
+    Using ``union`` as class method:
+
+    >>> from weldx import Time
+    >>> t1 = Time(["1s", "3s", "4s"])
+    >>> t2 = Time(["2s", "4s", "5s"])
+    >>>
+    >>> all(Time.union([t1, t2]) == Time(["1s", "2s", "3s", "4s", "5s"]))
+    True
+
+    Using the instance method:
+
+    >>> all(t1.union([t2]) == Time(["1s", "2s", "3s", "4s", "5s"]))
+    True
+
+    """
```

Comment:
I tested the examples in another docstring and they run. Docs are also rendering well, but we should keep this in mind.